test/spec.index.js

```diff
@@ -388,6 +388,40 @@ describe('Template Mixins', function () {
                 }));
             });
 
+            describe('form-group-year class', function () {
+                var dayCall;
+                var monthCall;
+                var yearCall;
+
+                beforeEach(function () {
+                    middleware(req, res, next);
+                    res.locals['input-date']().call(res.locals, 'field-name');
+
+                    render.called;
+
+                    dayCall = render.getCall(2);
+                    monthCall = render.getCall(4);
+                    yearCall = render.getCall(6);
+                });
+
+                it('not set with day', function () {
+                    dayCall.should.not.have.been.calledWith(sinon.match({
+                        formGroupClassName: 'form-group-year'
+                    }));
+                });
+                it('not set with month', function () {
+                    monthCall.should.not.have.been.calledWith(sinon.match({
+                        formGroupClassName: 'form-group-year'
+                    }));
+                });
+                it('not set with month', function () {
+                    yearCall.should.have.been.calledWith(sinon.match({
+                        formGroupClassName: 'form-group-year'
+                    }));
+                });
+
+            });
```

Comment:
"test one thing at a time" is sound advice, but it can be interpreted in many different ways. My personal feeling is that "one thing" should mean "can I precisely express the scope of the test in a single (short) sentence?". 

There are others who interpret this as "only ever one assertion per test case". These people are wrong.

Once you start splitting things up beyond the level I described you can easily end up writing test descriptions which describe the assertions and not the functionality itself, and so provide a worse description of the functionality. 

I prefer to start by writing a description which describes the thing being tested _as you would describe it out loud to another person_ and then writing whatever assertions you need to be able to describe that.